File: 3A/sortowanie.py

def sort_bubble(lista, n):
    for i in range(n - 1, 0, -1):
        for j in range(i):
            if lista[j] > lista[j + 1]:
                # tmp = lista[j]
                # lista[j] = lista[j + 1]
                # lista[j + 1] = tmp
                lista[j], lista[j + 1] = lista[j + 1], lista[j]
    return lista


def sort_insert(lista, n):
    for i in range(1, n):
        el = lista[i]
        j = i - 1
        while j > -1 and lista[j] > el:
            lista[j + 1] = lista[j]
            j = j - 1
        pass


def main():
    lista = [6, 5, 3, 1, 8, 7, 2, 4]
    n = len(lista)
    # lista = sort_bubble(lista, n)
    lista = sort_insert(lista, n)
    print(lista)
    return 0


main()

File: 2BP4/cpp_1/sortowanie.cpp
/*
 * sortowanie.cpp
 */
#include <iostream>
using namespace std;

void wypelnij(int t[], int roz, int n) {
    srand(time(NULL));
    for(int i=0; i<roz; i++) {
        t[i] = rand() % (n + 1);
    }
}

void drukuj(int t[], int roz) {
    for(int i=0; i<roz; i++) {
        cout << t[i] << " ";
    }
    cout << endl;
}

void zamien(int t[], int i) {
    int tmp = t[i];
    t[i] = t[i+1];
    t[i+1] = tmp;
}

void sort_bubble(int t[], int roz) {
    int licznik = 0;
    for (int j = roz - 1; j > 0; j--) {
        licznik++;
        for (int i = 0; i < j; i++) {
            // cout << "(" << i << "," << i+1 << ") ";
            licznik++;
            if (t[i] > t[i+1]) {
                zamien(t, i);
            }
        }
        // cout << endl;
    }
    cout << "Licznik: " << licznik << endl;
}

int main(int argc, char **argv)
{
    int roz = 20;
    int maks = 100;
	int tab[roz];
    wypelnij(tab, roz, maks);
    drukuj(tab, roz);
    sort_bubble(tab, roz);
    drukuj(tab, roz);
	return 0;
}

